Update solr.cmd for round #1 migration to SolrCLI

Willdotwhite opened the pull request:
All arg handling will be going through SolrCLI in v10

Updated in round 1:
* create
* create_core
* create_collection
* delete
* healthcheck
* status


I'm still not able to run the BATS tests locally, so all of the testing for this has been before/after manual testing. Commands still worked as expected (e.g. creating/deleting cores, printing out per-command usage instructions), but I'm not confident enough in my understanding of the whole application yet that I'd consider my solo testing sufficient.

---

## Work remaining


~`status` is giving me trouble, when I pass it into `run_solrcli` I get stop getting the expected results from `solr.cmd status` (e.g.):~

Willdotwhite committed:
Update solr.cmd for round 1 migration to SolrCLI

All arg handling will be going through SolrCLI in v10

Updated in round 1:
* create
* create_core
* create_collection
* delete
* healthcheck

Not complete:
* status

Willdotwhite commented:
Clearly my local zookeeper wasn't very happy with me - reverted back to `.\solr.cmd start -e cloud` which worked, so can now test the SolrCloud options

Willdotwhite commented:
Everything seems to be working other than a mime type issue with the healthcheck; it's hitting the haiku instead of the actual configuration endpoint

![image](https://github.com/epugh/solr/assets/13312237/91b7ace5-358d-4083-9c9f-c3b67e59e53f)


```
ERROR - 2023-06-18 12:00:54.492; org.apache.solr.cli.HealthcheckTool; ERROR: org.apache.solr.client.solrj.impl.BaseHttpSolrClient$RemoteSolrException: Error from server at http://localhost:7574/solr/zktest_shard1_replica_n1/admin/in
fo/system?wt=javabin&version=2: Expected mime type in [application/octet-stream, application/vnd.apache.solr.javabin] but got text/html. <p>
  Searching for Solr?<br/>
  You must type the correct path.<br/>
  Solr will respond.
</p>
 when trying to reach: http://localhost:7574/solr/zktest_shard1_replica_n1/
```

mkhludnev reviewed (commented):
definitely unrelated 
`start` prints redundant line `usage: status` **OK** I've found you mention that above. 
start usage prints unnecessary double quotes in port arg comment.

mkhludnev commented:
After running non-cloud instance `create` suggestion output has odd `null`s. Not a big deal though
```
dev>bin\solr start -V -v
...
dev>bin\solr create -c test
WARNING: Using _default configset. Data driven schema functionality is enabled by default, which is
         NOT RECOMMENDED for production use.

         To turn it off:
            curl http://localhost:8983/solr/null/config -d '{"set-user-property": {"update.autoCreateFields":"false"}}'
         Or:
            bin/solr config -c null -p 8983 -action set-user-property -property update.autoCreateFields -value false

Created new core 'test'

mkhludnev commented:
Hi @epugh 
healthcheck does something wrong 
```
ERROR - 2023-06-22 20:28:12.886; org.apache.solr.cli.HealthcheckTool; ERROR: org.apache.solr.client.solrj.impl.BaseHttpSolrClient$RemoteSolrException: Error from server at http://localhost:8983/solr/test3_shard1_replica_n1/admin/info/system?wt=javabin&version=2: Expected mime type in [application/vnd.apache.solr.javabin, application/octet-stream] but got text/html. <p>
  Searching for Solr?<br/>
  You must type the correct path.<br/>
  Solr will respond.
</p>
 when trying to reach: http://localhost:8983/solr/test3_shard1_replica_n1/
{
```

mkhludnev commented:
https://github.com/mkhludnev/solr/commit/718c5ef2a21db29dc5eaeb5ca5528afb323cd542
Here's what I've found playing with solr.cmd. 
I see that @epugh fixed healthcheck already. Wondering if you make it work in standalone mode too.
I'm not able to raise a PR in @Willdotwhite repo. So, just have a look at the linked changes above.

epugh commented:
Okay, you might see a comment from me to the two of you, I didn't quite realize that this PR actually existed!   I am running the precommit check...    Seems like the work flow is to merge this PR directly into my branch, which then goes into `apache/solr`!

epugh commented:
I guess I should wait till this is marked "Ready for REview" before I merge ;-)

epugh commented:
Looks like precommit passed.

Willdotwhite commented:
Hi both! Apologies for the radio silence, been a busy few days. I'll update this PR and put it for review on Sunday if that's ok, but I can add you both to this repo if you prefer :)

epugh commented:
No worries...  whichever works for you!  super excited about this contribution!

epugh commented:
Realizing I should have commented here..   @Willdotwhite since the `status` command was giving some problems, what if I back out my changes to `status` and then we could commit the good work you have done here?    BTW, how do you want to be credited in the `CHANGES.txt` file?

Willdotwhite commented:
That would make sense! I'll keep looking at `status`, but if we can move this slice forwards first then it'd be good to get some value added 👍🏻 

'Will White' works best for me, thanks :D 

(Apologies for missing the beat on Sunday, turns out an ordination takes a lot more time out of your weekend than I'd realised!)

Willdotwhite commented:
I've not had a go with 718c5ef2a21db29dc5eaeb5ca5528afb323cd542 yet, @mkhludnev do you want me to bring your commit into this PR or would you prefer to open a PR directly after this PR gets merged?

epugh commented:
Thanks @Willdotwhite ...   We're all volunteers ;-)

Willdotwhite commented:
Just as long as no-one sitting at their machine thinking "Damn, I'm sick of waiting for this @Willdotwhite guy"! 😆

mkhludnev commented:
> I've not had a go with [718c5ef](https://github.com/epugh/solr/commit/718c5ef2a21db29dc5eaeb5ca5528afb323cd542) yet, @mkhludnev do you want me to bring your commit into this PR or would you prefer to open a PR directly after this PR gets merged?

Hi, @Willdotwhite 
I suppose that the former approach (bring altogether) is easier for everyone.
FWIW, it seems that healthcheck is fixed in `main` already, so just ignore my changes in HealthcheckTool.

Willdotwhite committed:
Update solr.cmd status command output

Original source: 718c5ef2a21db29dc5eaeb5ca5528afb323cd542
curtesy of @mkhludnev

Manually copied ignoring existing fixes to `healthcheck`

Willdotwhite commented:
Yep, looks like the `status` changes are doing the beans 👍🏻 

```PS C:\Users\Will\Projects\solr\solr\packaging\build\dev\bin> .\solr.cmd status

Found Solr process 21424 running on port 7574
{                                                                                                         
  "solr_home":"C:\\Users\\Will\\Projects\\solr\\solr\\packaging\\build\\dev\\example\\cloud\\node2\\solr",
  "version":"10.0.0-SNAPSHOT 474afe50022dfbc9a63ed303ae4e60154ddd1816 [snapshot build, details omitted]", 
  "startTime":"Wed Jun 28 22:10:35 BST 2023",                                                             
  "uptime":"0 days, 0 hours, 1 minutes, 11 seconds",                                                      
  "memory":"319.7 MB (%62.4) of 512 MB",                                                                  
  "cloud":{                                                                                               
    "ZooKeeper":"127.0.0.1:9983",                                                                         
    "liveNodes":"2",                                                                                      
    "collections":"1"}}                                                                                   


Found Solr process 18996 running on port 8983
{                                                                                                         
  "solr_home":"C:\\Users\\Will\\Projects\\solr\\solr\\packaging\\build\\dev\\example\\cloud\\node1\\solr",
  "version":"10.0.0-SNAPSHOT 474afe50022dfbc9a63ed303ae4e60154ddd1816 [snapshot build, details omitted]", 
  "startTime":"Wed Jun 28 22:10:30 BST 2023",                                                             
  "uptime":"0 days, 0 hours, 1 minutes, 17 seconds",                                                      
  "memory":"86.1 MB (%16.8) of 512 MB",                                                                   
  "cloud":{
    "ZooKeeper":"127.0.0.1:9983",
    "liveNodes":"2",
    "collections":"1"}}
```

And I'm no longer seeing odd `usage: status` bits in the output of other commands.

Willdotwhite commented:
Thanks again @mkhludnev - I couldn't get a `git cherry-pick` working, so I copied across the changes manually and referenced your commit in the message, hope that's ok!
